pkg/storage/watch_cache.go

```diff
@@ -240,8 +245,6 @@ func (w *watchCache) updateCache(resourceVersion uint64, event *watchCacheEvent)
 
 // List returns list of pointers to <storeElement> objects.
 func (w *watchCache) List() []interface{} {
-	w.RLock()
-	defer w.RUnlock()
```

Comment:
the unpaired `defer w.RUnlock()` calls in WaitUntilFreshAndList and WaitUntilFreshAndGet look really weird... are those correct?